.github/workflows/windows.yml

```diff
@@ -15,3 +15,38 @@ jobs:
         cd build
         cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_VERBOSE_MAKEFILE=ON -DAMReX_BUILD_TUTORIALS=ON -DAMReX_FORTRAN=OFF -DAMReX_MPI=OFF
         cmake --build . --config Debug
+
+  # Build libamrex and all tutorials with CUDA
+  tutorials-cuda:
+    name: MSVC C++17 w/ CUDA@11.1.1 w/o Fortran w/o MPI
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+    - name: Windows CUDA Cache
+      uses: actions/cache@v2
+      with:
+        path: "C:/Program Files/NVIDIA GPU Computing Toolkit/"
+        key: msvc_cxx17_cuda111
+    - name: Download CUDA
+      run: |
+        if (Test-Path "C:\Program Files\NVIDIA GPU Computing Toolkit\") { exit }
+        New-item -ItemType directory -Name cuda
+        Set-Location -Path cuda -PassThru
+        Invoke-WebRequest -Uri 'https://developer.download.nvidia.com/compute/cuda/11.1.1/network_installers/cuda_11.1.1_win10_network.exe' -OutFile 'cuda_install.exe'
+        Start-Process -FilePath '.\cuda_install.exe' -ArgumentList '-s' -Wait -NoNewWindow
+    - name: Build & Install
+      shell: bash
+      run: |
+        export CUDA_PATH="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.1"
+        export CUDA_HOME=$CUDA_PATH
+        export PATH=$CUDA_PATH/bin:$PATH
+        export PATH="C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/:$PATH"
+        export CUDACXX="$CUDA_PATH/bin/nvcc.exe"
+        export CC="C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe"
+        export CXX="C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe"
+        mkdir build
+        cd build
+        cmake -G "Visual Studio 16 2019" -A x64 -DAMReX_BUILD_TUTORIALS=ON -DAMReX_FORTRAN=OFF -DAMReX_MPI=OFF -DAMReX_GPU_BACKEND=CUDA -DCMAKE_CXX_STANDARD=17 -DCMAKE_CUDA_STANDARD=17 -DAMReX_CUDA_ARCH=6.0 -DCUDAToolkit_ROOT="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.1" ..
```

Comment:
@robertmaynard this still does not find the `CMAKE_CUDA_COMPILER` - should one set more hints on Windows?

Reply:
The visual studio generator required the MSBuild extensions for CUDA to be installed, is that the case here? You might also need to specify the location of `CMAKE_CUDA_COMPILER`.

You might be better off trying the Ninja/Makefile generator as it doesn't require the MSBuild extensions and will have faster turnaround times.

Reply:
Thanks! I think that MSBuild should be here, how can one check? What's the CMake call you used to build AMReX with CUDA?

I thought that passing the `CUDAToolkit_ROOT` variable might indicate where `nvcc` is located, is that not the hint one would pass?

Reply:
Using CMake 3.19.1 from a normal shell the following was sufficient (For Visual Studio 2019 Generator):
```
cmake -S amrex/  -b <build_dir> \
   -DAMReX_GPU_BACKEND=CUDA \
   -DAMReX_FORTRAN=OFF \
   -DAMReX_MPI=ON \
   -DBUILD_SHARED_LIBS=OFF
```

For the Ninja generator I used:

```
cmake -S src -B nbuild -G Ninja -DAMReX_GPU_BACKEND=CUDA -DAMReX_FORTRAN=OFF -DAMReX_MPI=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_CUDA_COMPILER="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.1/bin/nvcc.exe"
```

In general if the CUDA msbuild extensions are installed, you should get nvcc detected automatically when using the Visual Studio Generator. IIRC `CUDACXX` might be ignored for the Visual Studio Generator and instead you would need to specify which CUDA version to use via `CMAKE_GENERATOR_TOOLSET` ( https://cmake.org/cmake/help/v3.19/variable/CMAKE_GENERATOR_TOOLSET.html )


```

Reply:
Thank you, which MPI did you use on Windows? Microsoft MPI?

Reply:
Yes, MS-MPI others should work but I didn't have those installed already.

Reply:
@robertmaynard I tested all of those options and none seem to work :(

@mic84 would work on #1639 once we get the setup here together (we have no windows at hand to test it without this CI setup)